src/java.base/linux/classes/sun/nio/fs/LinuxFileStore.java

```diff
@@ -141,29 +141,7 @@ public boolean supportsFileAttributeView(Class<? extends FileAttributeView> type
                 return false;
             }
 
-            // user_{no}xattr options not present but we special-case ext3 as
-            // we know that extended attributes are not enabled by default.
-            if (entry().fstype().equals("ext3")) {
-                return false;
-            }
-
-            // user_xattr option not present but we special-case ext4 as we
-            // know that extended attributes are enabled by default for
-            // kernel version >= 2.6.39
-            if (entry().fstype().equals("ext4")) {
-                if (!xattrChecked) {
-                    // check kernel version
-                    int[] kernelVersion = getKernelVersion();
-                    xattrEnabled = kernelVersion[0] > 2 ||
-                        (kernelVersion[0] == 2 && kernelVersion[1] > 6) ||
-                        (kernelVersion[0] == 2 && kernelVersion[1] == 6 &&
-                            kernelVersion[2] >= 39);
-                    xattrChecked = true;
-                }
-                return xattrEnabled;
-            }
-
-            // not ext3/4 so probe mount point
+            // no other information available so probe mount point
```

Comment:
You've removed the special handling for ext4 too, did you mean to do that? I think we'll need to do a lot of testing with this change as I suspect it going to bite back in some unusual configurations.

Reply:
Yes the change was intentional. I thought that ext4 has the same issues as ext3 and it might be possible to change the mount options or default mount options for ext4 with tune2fs to contain nouser_xattr. In this case the JDK would not detect that  extended attributes are not supported. But I must admit that I just falsified my theory. tune2fs does not allow nouser_xattr as mount option. So for ext4 it can only be that someone uses a kernel that was configured to not support extended attributes on ext4 - this might not occur in real life.
I will revert the change back to the original first change.